### PR TITLE
+SemiSupervised model and test file

### PR DESCRIPTION
**Pull request by BapProust**

### CODE REVIEW

File: models/SemiSupervisedModel.py [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+        # Initialize the SVM classifier
+        self.svm = SVC(kernel='linear', probability=True)
+
+    def train(self):
+        """
+        Train the semi-supervised model.
+        """
+        # Use LabelPropagation to generate pseudo-labels for the unlabeled data
+        print("Training LabelPropagation...")
+        self.label_propagation.fit(
```
Comment (HerySon) [inline models/SemiSupervisedModel.py:59]: Ton implémentation n'est pas correcte : LabelPropagation doit fonctionner en prenant en entrée tout le dataframe à entrainer en ayant laleblisé à -1 au préalable les valeurs inconnues de la target (voir la [doc correspondante de sckit-learn](https://scikit-learn.org/stable/modules/generated/sklearn.semi_supervised.LabelPropagation.html#sklearn.semi_supervised.LabelPropagation.fit))

File: models/SemiSupervisedModel.py [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+        # Combine the labeled and pseudo-labeled data
+        self.data = pd.concat([self.labeled_data, self.unlabeled_data])
+
+        print("Fitting SVM on combined data...")
+
+        # Fit the SVM classifier on the combined data
+        self.svm.fit(self.data[self.feature_cols], self.data[self.target_col])
+        print("Training complete.")
+
+    def evaluate(self):
```
Comment (HerySon) [inline models/SemiSupervisedModel.py:75]: L'usage de la fonction classification report est intéressante, mais pas très pertinente ici : il serait plus pertinent d'évaluer la qualité des labels prédits par `LabelPropagation`, par exemple avec une métrique de clustering

**Review by HerySon — CHANGES_REQUESTED** [review]

**Comment by HerySon** [comment]
@BapProust 

Comme spécifié dans les commentaires, ton implémentation pour entraîner `LabelPropagation` n'est pas correcte

## todo
- [ ] retravailler ta fonction pour que l'usage de `LabelPropagation` soit conforme à celle de la doc de scikit-learn
